modyn/supervisor/internal/triggers/datadrifttrigger.py

```diff
@@ -50,122 +49,145 @@ def init_trigger(self, context: TriggerContext) -> None:
-
-        The sliding window is determined by two pointers: detection_idx_start and detection_idx_end.
-        We hold the start pointer constant and advance the end pointer by detection_interval in every iteration.
-        In every iteration we run data drift detection on the sliding window of current data.
-
-        Note: When having remaining untriggered data from the previous batch of new data,
-        we include all of them in the first drift detection.
-
-        The remaining untriggered data has been processed in the previous new data batch,
-        so there's no need to run detection only on remaining data in this batch.
-        If a retraining is triggered, all data in the sliding window becomes triggering data.
-        Advance the start ptr after traversing the data_cache, we remove all the triggered data from the cache
-        and record the number of remaining untriggered samples.
-
-        Args:
-            new_data: List of new data (keys, timestamps, labels)
-            log: The log to store the trigger policy evaluation results to be able to verify trigger decisions
+        drift_eval_log = DriftTriggerEvalLog(
+            detection_idx_start=self._current_window[0][1],
+            detection_idx_end=self._current_window[-1][1],
+            triggered=triggered,
+            trigger_index=-1,
+            drift_results=drift_results,
+        )
 
-        Returns:
-            a generator here that waits for the previous trigger to finish and get the model.
-        """
-        # add new data to data_cache
-        self.data_cache.extend(new_data)
-
-        unvisited_data_points = len(self.data_cache)
-        untriggered_data_points = unvisited_data_points
-
-        # the sliding window of data points for detection
-        detection_idx_start = 0
-        detection_idx_end = 0
-
-        while unvisited_data_points >= self.config.detection_interval_data_points:
-            unvisited_data_points -= self.config.detection_interval_data_points
-            detection_idx_end += self.config.detection_interval_data_points
-            if detection_idx_end <= self.leftover_data_points:
-                continue
-
-            if self.previous_trigger_id is None:
-                triggered = True  # if no previous trigger exists, always trigger retraining
-                drift_results: dict[str, MetricResult] = {}
-            else:
-                # if exist previous trigger, detect drift
-                triggered, drift_results = self._run_detection(detection_idx_start, detection_idx_end)
-
-            drift_eval_log = DriftTriggerEvalLog(
-                detection_idx_start=detection_idx_start,
-                detection_idx_end=detection_idx_end,
-                triggered=triggered,
-                trigger_index=-1,
-                drift_results=drift_results,
+        if triggered:
+            self._reference_window = self._current_window  # Current assumption: same windowing strategy on both
```

Comment:
Totally fine for now, if I find the time for that, I'll update that after the deadline 👍